presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -1339,6 +1339,13 @@ public void testMergeEmptyHll()
         assertQuery("select cardinality(merge(empty_approx_set(0.1))) from orders");
     }
 
+    @Test
+    public void testStructureTypeData()
```

Comment:
Nit : Rename function to testStructFieldAccess()

Reply:
@kevinwilfong Kevin already added testDereference in #20727. This is not needed.

Reply:
@mbasmanova : We had found another problematic SQL in our tests `SELECT r[2] FROM (VALUES (ROW (ROW (1, 'a', true)))) AS v(r)`
@pdabre12 : The above SQL is what we want in the test. You can add it in the same test as Kevin in https://github.com/prestodb/presto/pull/20727